docs/introduction/create_custom_env.md

```diff
@@ -0,0 +1,222 @@
+```
+
+## Reset function
+
+```{eval-rst}
+.. py:currentmodule:: gymnasium.Env
+
+As purpose :meth:`reset` is to initiate a new episode for an environment and has two parameters: ``seed`` and ``options``. The seed can be used to initialize the random number generator to a deterministic state and options can be used to specify values used within reset. On the first line of the reset, you need to call ``super().reset(seed=seed)`` which will initialize the random number generate (:attr:`np_random`) to use through the rest of the :meth:`reset`.
+
+Within our custom environment, the :meth:`reset` needs to randomly choose the agent's and target positions (we repeat this if they have the same position). The return type of :meth:`reset` is a tuple of the initial observation and any auxiliary information. Therefore, we can use the methods ``_get_obs`` and ``_get_info`` that we implemented earlier for that:
+```
+
+```python
+    def reset(self, seed: Optional[int] = None, options: Optional[dict] = None):
+        # We need the following line to seed self.np_random
+        super().reset(seed=seed)
+
+        # Choose the agent's location uniformly at random
+        self._agent_location = self.np_random.integers(0, self.size, size=2, dtype=int)
+
+        # We will sample the target's location randomly until it does not coincide with the agent's location
+        self._target_location = self._agent_location
+        while np.array_equal(self._target_location, self._agent_location):
+            self._target_location = self.np_random.integers(
+                0, self.size, size=2, dtype=int
+            )
+
+        observation = self._get_obs()
+        info = self._get_info()
+
+        return observation, info
+```
+
+## Step function
+
+```{eval-rst}
+.. py:currentmodule:: gymnasium.Env
+
+The :meth:`step` method usually contains most of the logic for your environment, it accepts an ``action`` and computes the state of the environment after the applying the action, returning a tuple of the next observation, the action's reward, if the environment has terminated, if the environment has truncated and auxiliary information.
+```
+```{eval-rst}
+.. py:currentmodule:: gymnasium
+
+For our environment, we can use the ``self._action_to_direction`` to convert the discrete action (e.g., 2) to the grid direction with our agent location. However, to prevent our agent moving ourside the grid's bounds then we clip the agent's location to within the bounds. With the updated environment state from the action, we can compute the agent's reward and if the environment has terminated (the agent's location is equal to the target's location). As the environment doesn't truncate internally (we can apply a time limit wrapper to the environment during :meth:`make`) then this is permantently ``False``. To gather ``observation`` and ``info``, we can again make  use of ``_get_obs`` and ``_get_info``:
```

Comment:
I feel like this can be structured a bit better for readability:
"""
Several things need to happen during the step function:
1. We use the ``self._action_to_direction`` to convert the discrete action (e.g., 2) to a grid direction with our agent location. To prevent the agent from going out of bounds of the grd, we clip the agen't location to stay within bounds.
2. We compute the agent's reward by checking if the agent's current position is equal to the target's location.
3. Since the environment doesn't truncate internally (we can apply a time limit wrapper to the environment during :meth:`make`), we permanently set ``truncated`` to ``False``.
4. We once again use ``_get_obs`` and ``_get_info`` to obtain the agent's observation and auxiliary information.

"""